### ISSUE TITLE
DL3045: false positive, missing documentation

### ISSUE THREAD

**Issue by sfuerte** [hadolint/hadolint#560]
- [x] This is a bug report
- [ ] This is a feature request
- [x] I searched existing issues before opening this one

as a side note, didn't realize your Docker `latest` is linked to the Master branch versus the latest tagged release. 
Got broken pipelines because of https://github.com/hadolint/hadolint/commit/a556eafc42b10d188e098404e6ad617f90290f73

### Expected behavior

having locally:
```sh
> hadolint --version
Haskell Dockerfile Linter 1.23.0-no-git

> hadolint Dockerfile*

# i.e. all is fine
>
```

### Actual behavior

While our pipeline fails with the following:

```sh
<1s
+ hadolint --version
Haskell Dockerfile Linter v1.23.0-1-ga556eaf

hadolint Dockerfile*
<1s
+ hadolint Dockerfile*
Dockerfile:123 DL3045 warning: `COPY` to a relative destination without `WORKDIR` set.
Dockerfile:124 DL3045 warning: `COPY` to a relative destination without `WORKDIR` set.
Dockerfile:126 DL3045 warning: `COPY` to a relative destination without `WORKDIR` set.
Dockerfile:127 DL3045 warning: `COPY` to a relative destination without `WORKDIR` set.
Dockerfile:128 DL3045 warning: `COPY` to a relative destination without `WORKDIR` set.
Dockerfile:129 DL3045 warning: `COPY` to a relative destination without `WORKDIR` set.
...
```

![image](https://user-images.githubusercontent.com/16941995/109852619-2e1f7080-7c12-11eb-8d3d-508b67836df0.png)

Also, https://github.com/hadolint/hadolint/wiki/DL3045 doesn't exist and link is broken in README

### Steps to reproduce the behavior

**Dockerfile**

```Dockerfile
...

# Copy artefacts - lines 122-129
COPY --chown=${JETTY_USER}:${JETTY_USER} --from=jetty "${JETTY_BASE}" "${JETTY_BASE}"
COPY --chown=${JETTY_USER}:${JETTY_USER} --from=jetty /usr/local/jetty "${JETTY_HOME}"
COPY --chown=${JETTY_USER}:${JETTY_USER} --from=jetty docker-entrypoint.sh generate-jetty-start.sh /
COPY --chown=${JETTY_USER}:${JETTY_USER} --from=build /var/tmp/my_app/ "${JETTY_BASE}/webapps/${DEPLOYMENT_PREFIX}/"
COPY --chown=${JETTY_USER}:${JETTY_USER} my_app/ "${USER_HOME}/"
COPY --chown=${JETTY_USER}:${JETTY_USER} jetty/ "${JETTY_BASE}/"
COPY --chown=${JETTY_USER}:${JETTY_USER} webapps/ROOT/WEB-INF/ "${JETTY_BASE}/webapps/${DEPLOYMENT_PREFIX}/WEB-INF/"

# Container runtime settings
# reset back to Jetty user
USER ${JETTY_USER}
WORKDIR ${JETTY_BASE}
...
```

while the final `WORKDIR` is set after the `COPY` steps, but `hadolint` fails to recognize variables and whether they are using full or relative paths in them. In our case, all vars are with absolute paths.

**Comment by fk-windy** [comment]
Same thing, getting DL3045 while destination path is absolute

**Comment by m-ildefons** [comment]
I'm sorry to see that this broke your pipeline.
Hadolint does not expand environment variables in this context, so it has no clue whether or not  `${JETTY_BASE}` is in fact absolute.

@fk-windy , could you please elaborate? Does this happen with the destination path being an environment variable, or with a 'cleartext' absolute path?

I suppose there would be no harm in changing the rule not to trigger when the destination is an environment variable, though that would place responsibility on the programmer.

I've updated the wiki and will be making a fix soon.

**Comment by fk-windy** [comment]
> @fk-windy , could you please elaborate? Does this happen with the destination path being an environment variable, or with a 'cleartext' absolute path?

varible, like so
`COPY . $WORKDIR`

**Comment by GeoffCraig** [comment]
This happens if the destination is enclosed in double quotes.

COPY myfile.txt "/tmp/myfile.txt" == FAIL

COPY myfile.txt /tmp.myfile.txt == PASS

**Comment by sfuerte** [comment]
Works now. Thanks for resolving it quickly!

![image](https://user-images.githubusercontent.com/16941995/110041612-98fba500-7d01-11eb-9854-78274f1a1949.png)
